api/base/renderers.py

```diff
@@ -23,14 +23,14 @@ def render(self, data, accepted_media_type=None, renderer_context=None):
         # See JSON-API documentation on meta information: http://jsonapi.org/format/#document-meta
         data_type = type(data)
         if renderer_context is not None and data_type != str and data is not None:
-            meta_dict = renderer_context.get('meta')
+            meta_dict = renderer_context.get('meta', {})
             version = getattr(renderer_context['request'], 'version', None)
-            if meta_dict is not None:
-                if version:
-                    meta_dict['version'] = renderer_context['request'].version
-                    data.setdefault('meta', {}).update(meta_dict)
-            elif version:
-                meta_dict = {'version': renderer_context['request'].version}
+            warning = renderer_context['request'].META.get('warning', None)
+            if version:
+                meta_dict['version'] = renderer_context['request'].version
```

Comment:
Both of these can use the `version` value that was defined above!  meta_dict['version'] = version